engine/grid.py

```diff
@@ -253,44 +265,51 @@ def area(x1, y1, x2, y2, x3, y3):
 
         # Check if sum of A1, A2 and A3
         # is same as A
-        if (A == A1 + A2 + A3):
-            return True
+        return A == (A1 + A2 + A3)
+
+    def activate_line(self, row, col, n, isHorizontal):
+        """
+        Activate a single horizontal line of length n starting at row, col.
+        If isHorizontal is True, the line is from (row, col) to (row, col + n). 
+        Otherwise, the line is from (row, col) to (row + n, col). 
+        """
+        # Note: rows are y-position and columns are x-position
+        if isHorizontal:
+            for i in range(n):
+                self.nodes[col+i][row].is_active = True
         else:
-            return False
+            for i in range(n):
+                self.nodes[col][row+i].is_active = True
 
+    ##Activate triangle based on three points
     def activate_triangle(self, x1, y1, x2, y2, x3, y3):
         """
-        Activates all the nodes in a traingle based on three points.
+        Activates all the nodes in a traingle.
         """
-        rows = self.nodes.shape[0]
-        cols = self.nodes.shape[0]
-        for x in range(rows):
-            for y in range(cols):
-                if self.is_inside_triangle(x1, y1, x2, y2, x3, y3, x, y):
+        for x in range(self.num_rows):
+            for y in range(self.num_cols):
+                if self.isInsideTriangle(x1, y1, x2, y2, x3, y3, x, y):
                     self.nodes[x, y].is_active = True
 
-    
+    ##Checks if node is on border of activated nodes
     def is_on_border(self, row, col, row_limit, col_limit):
-        """
-        Checks if node is on border of activated nodes.
```

Comment:
I think spec would be nice. I'm assuming row and col is coor of node? Might make more sense to curry it into point and in the body of the code do (row, col) = point. Not sure about syntax of prev sentence, but you get the point

Reply:
good points. Also there are a bunch of other functions which also have row and col, so we can prob curry that into points as well

Reply:
good point from allen, lets do this in a separate pr!!